File: SportsSideKick/app/src/main/java/base/app/ui/fragment/content/NewsFragment.java
package base.app.ui.fragment.content;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.orangegangsters.github.swipyrefreshlayout.library.SwipyRefreshLayout;
import com.orangegangsters.github.swipyrefreshlayout.library.SwipyRefreshLayoutDirection;
import com.wang.avi.AVLoadingIndicatorView;

import org.greenrobot.eventbus.Subscribe;

import java.util.List;

import base.app.R;
import base.app.data.news.NewsModel.NewsType;
import base.app.ui.adapter.content.NewsAdapter;
import base.app.ui.fragment.base.BaseFragment;
import base.app.data.news.NewsModel;
import base.app.data.news.NewsPageEvent;
import base.app.data.wall.WallNews;
import butterknife.BindView;
import butterknife.ButterKnife;

import static base.app.data.news.NewsModel.NewsType.*;

/**
 * Created by Djordje on 12/29/2016.
 * Copyright by Hypercube d.o.o.
 * www.hypercubesoft.com
 */
public class NewsFragment extends BaseFragment {

    NewsType type = OFFICIAL;
    NewsAdapter adapter;
    @BindView(R.id.swipeRefreshLayout)
    SwipyRefreshLayout swipeRefreshLayout;
    @BindView(R.id.recyclerView)
    RecyclerView recyclerView;
    @BindView(R.id.progressBar)
    AVLoadingIndicatorView progressBar;
    @BindView(R.id.topImage)
    ImageView topImage;
    @BindView(R.id.topСaption)
    TextView topCaption;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_news, container, false);
        ButterKnife.bind(this, view);

        Glide.with(getContext()).load(R.drawable.image_wall_background).load(topImage);

        adapter = new NewsAdapter();

        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setNestedScrollingEnabled(false);

        List<WallNews> existingItems = NewsModel.getInstance().getAllCachedItems(type);
        if (existingItems!=null && existingItems.size() > 0)
        {
            adapter.getValues().addAll(existingItems);
            progressBar.setVisibility(View.GONE);
        }
        else {
            NewsModel.getInstance().loadPage(type);
        }
        swipeRefreshLayout.setOnRefreshListener(new SwipyRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh(SwipyRefreshLayoutDirection direction) {
                NewsModel.getInstance().setLoading(false,type);
                NewsModel.getInstance().loadPage(type);
            }
        });
        return view;
    }

    @Subscribe
    public void onNewsReceived(NewsPageEvent event) {
        progressBar.setVisibility(View.GONE);
        swipeRefreshLayout.setRefreshing(false);
        adapter.getValues().addAll(event.getValues());
        adapter.notifyDataSetChanged();
    }
}

File: SportsSideKick/app/src/main/java/base/app/ui/fragment/content/NewsFragment.kt
package base.app.ui.fragment.content

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.OvershootInterpolator
import base.app.R
import base.app.data.news.NewsModel
import base.app.data.news.NewsModel.NewsType.OFFICIAL
import base.app.data.news.NewsPageEvent
import base.app.data.wall.WallNews
import base.app.ui.adapter.content.NewsAdapter
import base.app.ui.fragment.base.BaseFragment
import base.app.util.events.post.ItemUpdateEvent
import base.app.util.ui.inflate
import base.app.util.ui.show
import jp.wasabeef.recyclerview.animators.SlideInUpAnimator
import kotlinx.android.synthetic.main.fragment_news.*
import org.greenrobot.eventbus.Subscribe

/**
 * Created by Djordje on 12/29/2016.
 * Copyright by Hypercube d.o.o.
 * www.hypercubesoft.com
 */
open class NewsFragment : BaseFragment() {

    internal val type = OFFICIAL
    internal var adapter: NewsAdapter = NewsAdapter(OFFICIAL)

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, state: Bundle?): View? {
        return container?.inflate(R.layout.fragment_news)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        topImage.show(R.drawable.image_wall_background)

        recyclerView.adapter = adapter
        recyclerView.itemAnimator = SlideInUpAnimator(OvershootInterpolator(1f))

        loadNews()
    }

    private fun loadNews() {
        swipeRefreshLayout.isRefreshing = true
        val existingItems = NewsModel.getInstance().getAllCachedItems(type)
        if (existingItems != null && existingItems.size > 0) {
            adapter.addAll(existingItems)
            swipeRefreshLayout.isRefreshing = false
        } else {
            NewsModel.getInstance().loadPage(type)
        }
        swipeRefreshLayout.setOnRefreshListener {
            NewsModel.getInstance().setLoading(false, type)
            NewsModel.getInstance().loadPage(type)
        }
    }

    @Subscribe
    fun onNewsReceived(event: NewsPageEvent) {
        adapter.addAll(event.values)
        swipeRefreshLayout.isRefreshing = false
    }

    @Subscribe
    fun onItemUpdate(event: ItemUpdateEvent) {
        val news = event.post as WallNews
        adapter.values.forEachIndexed { index, item ->
            if (item.postId == news.postId) {
                adapter.remove(item)
                adapter.add(index, news)
                return
            }
        }
    }
}